lib/game.rb

```diff
@@ -0,0 +1,94 @@
+class Game
+  @@v_conditions = [[1,2,3], [4,5,6], [7,8,9], [1,4,7], [2,5,8], [3,6,9], [1,5,9], [3,5,7]]
+
+  def initialize(p1, p2, board)
+    @p1 = p1
+    @p2 = p2
+    @board = board
+    @active_player = rand()*10 < 5 ? @p2 : @p1
+  end
+
+  def cycle
+    loop do
+      change_active_player
+      get_input
+      save_input
+      show(@board)
+      break if victory?
+      break if tie?
+    end
+  end
+
+  def restart?
+    loop do
+      puts "\nPlay again? Y/N"
+      input = gets.chomp.upcase
+      if input == "Y" || input == "YES"
+        reset
+        return true
+      elsif input == "N" || input == "NO"
+        return false
+      else
+        puts "Invalid input. Please type Y or N."
+      end
+    end
+  end
+
+  private
+
+  def get_input
```

Comment:
This is a UI related logic and should not reside inside this class. You can simply make a module that has all the UI related code and then require it in this file and include it. For more information about modules please refer to the Odin lessons that cover this project